pkg/broker/filter/filter_handler.go

```diff
@@ -206,34 +223,65 @@ func (h *Handler) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
 
 	h.reportArrivalTime(event, reportArgs)
 
-	h.send(ctx, writer, request.Header, subscriberURI.String(), reportArgs, event, ttl)
+	h.send(ctx, writer, request.Header, subscriberURI.URL(), reportArgs, event, ttl)
 }
 
-func (h *Handler) send(ctx context.Context, writer http.ResponseWriter, headers http.Header, target string, reportArgs *ReportArgs, event *cloudevents.Event, ttl int32) {
+func (h *Handler) send(ctx context.Context, writer http.ResponseWriter, headers http.Header, target *url.URL, reportArgs *ReportArgs, event *cloudevents.Event, ttl int32) {
 	// send the event to trigger's subscriber
-	response, err := h.sendEvent(ctx, headers, target, event, reportArgs)
-	if err != nil {
-		h.logger.Error("failed to send event", zap.Error(err))
-		writer.WriteHeader(http.StatusInternalServerError)
-		_ = h.reporter.ReportEventCount(reportArgs, http.StatusInternalServerError)
+	response, responseErr := h.sendEvent(ctx, headers, target, event, reportArgs)
+
+	if responseErr.err != nil {
+		h.logger.Error("failed to send event", zap.Error(responseErr.err))
+		// If error is not because of the response, it should respond with http.StatusInternalServerError
+		if responseErr.ResponseCode == NoResponse {
+
+			writer.WriteHeader(http.StatusInternalServerError)
+			_ = h.reporter.ReportEventCount(reportArgs, http.StatusInternalServerError)
+			return
+		}
+		// If send err is not nil, it don't need to adds the specified HTTP Headers to the ResponseWriter
+		if !responseErr.sendErr {
+			proxyHeaders(response.Header, writer)
+		}
```

Comment:
Should this check be more like this one:
```suggestion
		// If error has a response propagate subscriber's headers back to channel
		if response != nil {
			proxyHeaders(response.Header, writer)
		}
```